django/db/migrations/operations/base.py

```diff
@@ -98,17 +98,15 @@ def references_field(self, model_name, name, app_label=None):
         """
         return self.references_model(model_name, app_label)
 
-    def allowed_to_migrate(self, connection_alias, model):
+    def allowed_to_migrate(self, connection_alias, model, hints=None):
         """
-        Returns if we're allowed to migrate the model. Checks the router,
-        if it's a proxy, if it's managed, and if it's swapped out.
+        Returns if we're allowed to migrate the model.
         """
-        return (
-            not model._meta.proxy and
-            not model._meta.swapped and
-            model._meta.managed and
-            router.allow_migrate(connection_alias, model)
-        )
+        # Always skip if proxy, unmanaged or swapped out.
+        if model and (model._meta.proxy or model._meta.swapped or not model._meta.managed):
+            return False
+
```

Comment:
You had a `if not model and not hints` here before. Any reason why this is gone? The only thing I can think of, routers have to take care of `model=None` anyways,  so this "precaution" isn't necessary anymore.